pydantic/fields.py

```diff
@@ -310,34 +311,50 @@ class MyModel(pydantic.BaseModel):
         Returns:
             An instance of the field metadata.
         """
-        final = False
-        if _typing_extra.is_finalvar(annotation):
-            final = True
-            if annotation is not typing_extensions.Final:
+        # 1. Check if the annotation is the `Final` type qualifier:
+        final = _typing_extra.is_finalvar(annotation)
+        if final:
+            if _typing_extra.is_generic_alias(annotation):
+                # The annotation is a parametrized `Final`, e.g. `Final[int]`.
```

Comment:
Maybe prefix with 1.1, like you've done below for #3? And same for else block below